diff -u src/admc/main.cpp src/admc/main.cpp
--- src/admc/main.cpp	2026-03-26 23:33:55.852042747 +0700
+++ src/admc/main.cpp	2026-03-26 23:37:42.015558322 +0700
@@ -117,7 +117,7 @@
         }
     }
     catch (const std::runtime_error& e) {
-        qWarning(e.what());
+        qWarning("%s", e.what());
     }

     load_connection_options();
diff -u src/admc/managers/icon_manager.cpp src/admc/managers/icon_manager.cpp
--- src/admc/managers/icon_manager.cpp	2026-03-25 21:22:18.000000000 +0700
+++ src/admc/managers/icon_manager.cpp	2026-03-26 23:36:57.515047160 +0700
@@ -317,7 +317,7 @@
 }

 QStringList IconManager::available_themes() {
-    const QStringList available_themes = {impl->system_theme, impl->custom_theme};
+    const QStringList available_themes = {impl->system_theme};

     return available_themes;
 }
